fix: update importer functions to match new names

smallid opened the pull request:
Rename importer interface functions file_account, file_date, and file_name to account, date, and filename respectively. Matches the new beangulp (beancount v3) interface names.

Renamed existing `self.date` to `self.date_value` to avoid symbol conflicts and interface confusion.

resolves #131

smallid commented:
I've done only minimal testing with my local setup - looks like some of the automated checks are failing so I'll follow up on that.

redstreet commented:
Sounds good. The failing checks seem to be the linter.

smallid commented:
Fixing the pytest failures required numerous file renames, and also removing several conflicting `def account(self, filename):` functions in specific importers.

For example I removed this function in `beancount_reds_importers/importers/schwab/schwab_csv_brokerage.py`
```
   def account(self, filename):
        return self.config["main_account"]
```

I didn't find any existing usage of these functions, and the base implementation in `libreader/reader.py` seems to be quite comprehensive (including removing placeholders like `{ticker}`).

@redstreet please take a look at those changes to see if they are acceptable. (I've kept the commit separate for now to make it easier to rework if needed)

redstreet commented:
Hmm, the renames should be unnecessary. Are you running pytest after rebasing to the top of the branch? Passes fine for me.

smallid commented:
I changed the test generator `beancount_reds_importers/util/regression_pytest.py` to look for files that correspond with the new test functions - I thought that would be less confusing than keeping the existing names. I can definitely revert that part of the change though - let me know.

Just rebased from latest main - pytest is passing for me with the current set of changes (SHA `b2c9920d`)

smallid commented:
@redstreet I think this PR is good to merge - all the checks seem to be passing. Let me know if I'm missing anything or you want me to make any changes.

smallid commented:
I renamed the instance variables - a couple to `max_date` and one to `paycheck_date_value` (since there was a conflicting function named `paycheck_date`)

redstreet commented:
Awesome, thanks much for this PR! Merging.
